File: CPP/cell.cpp
#include <iostream>
using namespace std ;

#include "cell.h"

void Cell::render() {
	if (!this->getCreation() ){
		cout << "n" ;
	}
	else
	{
		this->getCreation()->render() ;	
	}
}

void Cell::Interact(int i) {
	//Nothing
}

void Cell::setCreation(Creation* _c){
	creation = _c ;
}

Creation* Cell::getCreation(){
	return creation ;
}

int Cell::getIdentity() {
	return -1 ; //Cell Murni
}

bool Cell::GetAtribute(int i) {
	return false ;
}

void Cell::Update() {
	if (!this->getCreation() ){
		//Nothing
	}
	else
	{
		this->getCreation()->Action(1) ; // Update	
	}
}

bool Cell::IsPossibleToInteract(int i) {
	return false ;	
}

		

File: main/src/Cell.java
package main.src;
//#include "renderable.h"
//#include "creation.h"

public class Cell implements Renderable{

private Creation creation;

public void render() {
	if (this.getCreation() == null ){
		System.out.print( "n" );
	}
	else
	{
		this.getCreation().render() ;	
	}
}

public void interact(int i) {
	//Nothing
}

public void setCreation(Creation _c){
	creation = _c ;
}

public Creation getCreation(){
	return creation ;
}

public int getIdentity() {
	return -1 ; //Cell Murni
}

public boolean getAtribute(int i) {
	return false ;
}

public void update() {
	if (this.getCreation() == null){
		//Nothing
	}
	else
	{
		this.getCreation().Action(1) ; // Update	
	}
}

public boolean isPossibleToInteract(int i) {
	return false ;	
}

}
